int32 submap_index
---
int32[] connections
